lib/replicator.js

```diff
@@ -2161,6 +2180,16 @@ module.exports = class Replicator {
     return false
   }
 
+  _updateHotswap (peer) {
+    const maxHotswaps = peer.getMaxHotswapInflight()
+    if (!peer.isActive() || peer.inflight >= maxHotswaps) return
+
+    for (const b of this.hotswaps.pick(peer)) {
+      if (peer._requestBlock(b) === false) continue
+      if (peer.inflight >= maxHotswaps) break
```

Comment:
(sanity check) can ` _requestBlock` never result in additional inflights when it returns `false`? I'm tracing the `_maybeWant` path, and it's not trivial to figure out (`_maybeWant` can send a `wireWant` message, though I don't think that results in an `inflight` increment)

Reply:
only block requests count as inflight atm, we should prop rename post rocks to blocksInflight for clarity